src/components/useVerboseFormik.ts

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import { FormikValues, useFormik } from "formik";
+import {
+  FormFieldStates,
+  useFieldStates,
+  VerboseFieldState,
+} from "../lib/formValidationTypes";
+
+/**
+ * Extends the {@link useFormik} function by adding state property supporting
+ * verbose field state management.
+ *
+ * @param initialValues The initial values for the form fields.
+ * @param initialStates The initial states of the form fields.
+ *
+ * @returns formik object
+ */
+export default function useVerboseFormik<FormParameters extends FormikValues>({
```

Comment:
How's this "verbose" ? Perhaps `extended`?

Reply:
I used "verbose" since the extension allows the form to hold not only error but also warning, info and success messages and I used the "verbose" in `VerboseTextField` etc.

But I guess, it's not really that clear and I'll change it.

Reply:
Changed in https://github.com/inrupt/client-identifier-helper/pull/185/commits/87706799a5888292d73bfd9845e2ffca3dab863f